File: java-core-plugins/src/main/java/org/unlaxer/tramli/plugins/audit/AuditStorePlugin.java
package org.unlaxer.tramli.plugins.audit;

import org.unlaxer.tramli.FlowStore;
import org.unlaxer.tramli.plugins.api.PluginDescriptor;
import org.unlaxer.tramli.plugins.api.StorePlugin;

public final class AuditStorePlugin implements StorePlugin {
    private final ProducedDataSerializer serializer;

    public AuditStorePlugin() {
        this(new DefaultProducedDataSerializer());
    }

    public AuditStorePlugin(ProducedDataSerializer serializer) {
        this.serializer = serializer;
    }

    @Override
    public PluginDescriptor descriptor() {
        return new PluginDescriptor("audit-store", "Audit Store", "Captures produced-data diffs per transition.");
    }

    @Override
    public FlowStore wrapStore(FlowStore store) {
        return new AuditingFlowStore(store, serializer);
    }
}

File: ts-plugins/src/audit/audit-store-plugin.ts
import type { StorePlugin, PluginDescriptor } from '../api/types.js';
import { AuditingFlowStore } from './auditing-flow-store.js';

export class AuditStorePlugin implements StorePlugin {
  descriptor(): PluginDescriptor {
    return { id: 'audit', displayName: 'Audit Store', description: 'Captures transition + produced-data diffs' };
  }
  kind() { return 'STORE' as const; }

  wrapStore(store: any): AuditingFlowStore {
    return new AuditingFlowStore(store);
  }
}
